s3/pkg/datastore/azure/azure.go

```diff
@@ -0,0 +1,286 @@
+// Copyright 2019 The OpenSDS Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azure
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"encoding/binary"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	log "github.com/sirupsen/logrus"
+	backendpb "github.com/opensds/multi-cloud/backend/proto"
+	. "github.com/opensds/multi-cloud/s3/error"
+	dscommon "github.com/opensds/multi-cloud/s3/pkg/datastore/common"
+	"github.com/opensds/multi-cloud/s3/pkg/model"
+	pb "github.com/opensds/multi-cloud/s3/proto"
+)
+
+// TryTimeout indicates the maximum time allowed for any single try of an HTTP request.
+var MaxTimeForSingleHttpRequest = 50 * time.Minute
+
+type AzureAdapter struct {
+	backend      *backendpb.BackendDetail
+	containerURL azblob.ContainerURL
+}
+
+/*func Init(backend *backendpb.BackendDetail) *AzureAdapter {
+	endpoint := backend.Endpoint
+	AccessKeyID := backend.Access
+	AccessKeySecret := backend.Security
+	ad := AzureAdapter{}
+	containerURL, err := ad.createContainerURL(endpoint, AccessKeyID, AccessKeySecret)
+	if err != nil {
+		log.Infof("AzureAdapter Init container URL faild:%v\n", err)
+		return nil
+	}
+	adap := &AzureAdapter{backend: backend, containerURL: containerURL}
+	log.Log("AzureAdapter Init succeed, container URL:", containerURL.String())
+	return adap
+}*/
+
+func (ad *AzureAdapter) createContainerURL(endpoint string, acountName string, accountKey string) (azblob.ContainerURL,
```

Comment:
do we need versioning here ? like, this AzureAdapter is compatible with which version of the Azure API spec ?

Reply:
Hi @rhsakarpos the version of SDK is defined in Gopkg.toml, for Azure, it's 0.3.0 now.